app/components/Contacts/ContactForm/ContactForm.jsx

```diff
@@ -27,46 +41,168 @@ export default class ContactForm extends React.Component<Props> {
   }
 
   render() {
-    const { submitLabel, name, address } = this.props
+    const { submitLabel, formName, mode, formAddress } = this.props
+    const { nameError, addressError } = this.state
+
+    let heading = 'Add A Contact'
+    let subHeading = 'Insert Contact Details'
+
+    if (mode === 'edit') {
+      heading = 'Edit A Contact'
+      subHeading = 'Modify Contact Details'
+    }
 
     return (
-      <form className={styles.contactForm} onSubmit={this.handleSubmit}>
-        <TextInput
-          className={styles.input}
-          placeholder="Contact Name"
-          value={name}
-          onChange={this.handleChangeName}
-        />
-        <TextInput
-          className={styles.input}
-          placeholder="Address"
-          value={address}
-          onChange={this.handleChangeAddress}
-        />
-        <Button
-          className={styles.button}
-          primary
-          type="submit"
-          renderIcon={AddIcon}
-        >
-          {submitLabel}
-        </Button>
-      </form>
+      <section>
+        <h1 className={styles.contactFormHeader}>
+          <AddIcon /> {heading}
+        </h1>
+        <form className={styles.contactForm} onSubmit={this.handleSubmit}>
+          <h2 className={styles.contactFormSubheader}>{subHeading}</h2>
+          <label htmlFor="contactName" className={styles.contactFormLabel}>
+            Name
+          </label>
+          <TextInput
+            id="contactName"
+            name="name"
+            className={styles.input}
+            placeholder="Enter Contact Name..."
+            value={formName}
+            onChange={this.handleChangeName}
+            error={nameError}
+          />
+          <label htmlFor="contactAdress" className={styles.contactFormLabel}>
+            Wallet Address
+          </label>
+          <TextInput
+            id="contactAddress"
+            name="address"
+            className={styles.input}
+            placeholder="Enter Wallet Address..."
+            value={formAddress}
+            onChange={this.handleChangeAddress}
+            error={addressError}
+          />
+          <DialogueBox
+            icon={<WarningIcon />}
+            text="Please review and ensure that you have entered the address correctly to avoid loss of funds"
+            className={styles.conactFormDialogue}
+          />
+          <Button
+            className={styles.button}
+            primary
+            type="submit"
+            renderIcon={AddContactIcon}
+          >
+            {submitLabel}
+          </Button>
+        </form>
+      </section>
     )
   }
 
+  componentWillMount() {
+    const { newAddress, setAddress, address } = this.props
+
+    this.setState({ ownAddress: address })
```

Comment:
Cant the state object just be initialized with this value as opposed to being set in componentWillMount?